### ISSUE TITLE
[BUG] Reverse Proxy does not seem to work for playback (only queueing songs)

### ISSUE THREAD

**Issue by acamino84** [vicwomg/pikaraoke#431]
I have been playing around with PiKaraoke and reverse proxies and have made some progress but have hit a roadblock.

**Progress Made So Far**

Step 1: Followed install instructions for 1.6.0, running in --headless mode, confirmed it was working on local network, properly playing on web browser on other device on network. 

Step 2: Set up reverse proxy to point to the :5555 port of PiKaraoke device. Accessing domain externally properly connects to the queueing system of PiKaraoke, songs can be queued without issue. This helps users queue songs without having to connect to local network.

**Issue Encountered**

After the above, I tried to see if I could also have the /splash device play the videos from outside the local network. 

Step 3: The reverse proxy set up above (obviously) automatically redirects the /splash page to the PiKaraoke device. The confirmation popup appears and is properly satisfied. Adding a video from the queuing page (regardless if on local network or not) results in the text of the video being added to the remotely accessed /splash page _**appearing**_ as if the video is about to start, _**however**_, the video never starts. The terminal for PiKaraoke shows that the song ends immediately after starting.

Step 4: I tried forwarding port 5556 to PiKaraoke, this does not help.

Step 5: I tried using the "-m" flag with a custom subdomain "-m {custom ffmpeg subdomain}" flag and then setting up a reverse proxying that subdomain to the PiKaraoke 5556 port but that doesn't help either.

As of now, despite the above, I am unable to properly play videos on the /splash page outside of the local network.

Any thoughts?

**Comment by vicwomg** [comment]
Very hard to troubleshoot this without your URL specifics, but I understand your need for privacy.

Seems very likely your port 5556 proxying is not working. Open your browser's network inspector and sniff out the URL it's trying to access when starting a stream. Then figure out why that URL is not accessible externally.

**Comment by acamino84** [comment]
I fixed it! Your email put me on the right track, thank you! I was properly using the -m flag, but didn't include it in the startup on boot script. 

Happy to write a reverse proxy guide if you think it would help future users.

**Comment by vicwomg** [comment]
A guide would be very helpful, thanks!

**Comment by vicwomg** [comment]
@acamino84  I will close this ticket, but if you write up a guide and post it as a comment I'd be happy to add it to the wiki.

**Comment by fuentes81081** [comment]
@acamino84 I'm running into this issue as well.  A write-up would be great!

**Comment by vicwomg** [comment]
@fuentes81081  this ticket is closed because the original premise of running pikaraoke on multiple ports is not relevant anymore.  Can you file a ticket with the exact issue you're having? It's difficult to follow from the given context.

**Comment by acamino84** [comment]
I was going to write a guide on how to run it on a proxy, but not sure if it still works because it appears (from your comment) it doesnt run on various ports anymore?

Either way, here's my solution for running PiKaraoke on a proxy, unsure if it still works but here you go. I think that explaining the problem that I encountered might help explain how this proxy worked. 

So originally, I pointed Caddy (although I'm fairly certain that Apache and Nginx would also work as they handle proxies similarly).

**Step 1**
First, I realized that PiKaraoke ran on port 5555, so I pointed a subtdomain to PiKaraoke using the following Caddy instructions in the caddyfile:

`
karaoke.[my_domain] {

 reverse_proxy * {
                to [PiKaraoke_internal_ip]:5555
}
}
`

**Step 2**
The Splash screen and the song request screen worked well, but I then realized that videos were not working. I looked into this issue and noticed that the reason was because videos were being sent on port 5556 (a different port). So I added the following Caddy instruction in my caddy file as well:

`karaoke-ffmpeg.[my_domain], karaoke.[my_domain]:5556 {

 reverse_proxy * {
                to [PiKaraoke_internal_ip]:5556
}
}`

**Step 3**
The reason a added both the "karaoke-ffmpeg" subdomain as well as simply the domain with the port was because I wasnt sure which one would "solve" the reverse proxy issue I was having. As of the above two instructions, videos were still not being sent.

But then I realized that when an external browser on the splash screen was accessing the player, the videos would be sent to the 5556 external port but not the 5556 internal port. And (for hopefully obvious reasons) I did not wish to open an external port directly tot he pikaraoke server. So I did one more thing. I added the following flags/arguments to the Pi Karaoke command line:

`pikaraoke -u https://[my_domain] -m https://karaoke-ffmpeg.[my_domain] `

**Conclusion**
With the above three steps, you will notice that your reverse proxy should be working perfectly.

(I also added the "--headless" "--admin-password [password]" "--logo-path [path to logo jpg on pi]" flags, but they did not affect the reverse proxy's functioning, I just added them because those were options I selected)
